geyser-plugin-manager/src/geyser_plugin_service.rs

```diff
@@ -208,4 +221,384 @@ impl GeyserPluginService {
+        // Check if any plugin names match this one
+        let Some(idx) = plugin_manager.plugins.iter().position(|plugin| plugin.name().eq(name)) else {
+            // If we don't find one return an error
+            return Err(
+                jsonrpc_core::error::Error {
+                    code: ErrorCode::InternalError,
+                    message: String::from("plugin requested to unload is not loaded"),
+                    data: None,
+                }
+            )
+        };
+
+        // Unload and drop plugin
+        let mut current_plugin = plugin_manager.plugins.remove(idx);
+        current_plugin.on_unload();
+        drop(current_plugin);
+
+        Ok(())
+    }
+
+    /// Checks for a plugin with a given `name`.
+    /// If it exists, first unload it.
+    /// Then, attempt to load a new plugin
+    pub fn reload_plugin_rpc(
+        plugin_manager: &mut GeyserPluginManager,
+        name: &str,
+        config_file: &str,
+    ) -> JsonRpcResult<()> {
+        // Check if any plugin names match this one
+        let Some(idx) = plugin_manager.plugins.iter().position(|plugin| plugin.name().eq(name)) else {
+            // If we don't find one return an error
+            return Err(
+                jsonrpc_core::error::Error {
+                    code: ErrorCode::InternalError,
+                    message: String::from("plugin requested to reload is not loaded"),
+                    data: None,
+                }
+            )
+        };
+
+        // Unload and drop current plugin first in case plugin requires exclusive access to resource,
+        // such as a particular port or database.
+        let mut current_plugin = plugin_manager.plugins.remove(idx);
+        current_plugin.on_unload();
+        drop(current_plugin);
+
+        // Try to load plugin, library
+        // SAFETY: It is up to the validator to ensure this is a valid plugin library.
+        let mut new_plugin: Box<dyn GeyserPlugin> = load_plugin_from_config(Path::new(config_file))
+            .map_err(|err| jsonrpc_core::Error {
+                code: ErrorCode::InternalError,
+                message: err.to_string(),
+                data: None,
+            })?;
+
+        // Attempt to on_load with new plugin
+        match new_plugin.on_load(&config_file) {
+            // On success, replace plugin and library
+            Ok(()) => {
+                plugin_manager.plugins.push(new_plugin);
+            }
+
+            // On success, replace plugin and library
+            Err(err) => {
+                return Err(jsonrpc_core::error::Error {
+                    code: ErrorCode::InternalError,
+                    message: format!(
+                        "failed to start new plugin (previous plugin was dropped!): {err}"
+                    ),
+                    data: None,
+                });
+            }
+        }
+
+        Ok(())
+    }
+}
+
+#[cfg(not(test))]
+fn load_plugin_from_config(
```

Comment:
Put this code near where the load_plugin was to make it easier to review -- there is not much real change.